froide/foirequest/api/views/message.py

```diff
@@ -0,0 +1,55 @@
+from django.contrib.auth import get_user_model
+from django.utils.translation import gettext as _
+
+from django_filters import rest_framework as filters
+from rest_framework import permissions, viewsets
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.views import Request
+
+from froide.foirequest.api.permissions import OwnsFoiRequestPermission
+
+from ...auth import (
+    get_read_foimessage_queryset,
+)
+from ...models import FoiMessage
+from ..serializers import FoiMessageSerializer, optimize_message_queryset
+
+User = get_user_model()
+
+
+class FoiMessageFilter(filters.FilterSet):
+    class Meta:
+        model = FoiMessage
+        fields = ("request", "kind", "is_response", "is_draft")
+
+
+class FoiMessageViewSet(viewsets.ModelViewSet):
+    serializer_class = FoiMessageSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = FoiMessageFilter
+    required_scopes = ["make:message"]
+
+    def get_permissions(self):
+        if self.action in ["list", "retrieve"]:
+            return []
+        return [permissions.IsAuthenticated(), OwnsFoiRequestPermission()]
+
+    def get_queryset(self):
+        qs = get_read_foimessage_queryset(self.request).order_by()
+        return self.optimize_query(qs)
+
+    def optimize_query(self, qs):
+        return optimize_message_queryset(self.request, qs)
+
+    def ensure_draft(self):
+        # only drafts can be edited
+        if not self.get_object().is_draft:
+            raise PermissionDenied(_("Cannot alter non-draft messages"))
```

Comment:
This should probably be done using a permission-class. https://www.django-rest-framework.org/api-guide/permissions/#object-level-permissions